api/src/main/java/com/epam/pipeline/manager/datastorage/DataStorageManager.java

```diff
@@ -778,16 +779,19 @@ public List<DataStorageTagSearchResult> searchDataStorageItemByTag(
             final Map<Long, List<DataStorageTag>> results = new HashMap<>();
             final Map<Long, List<AbstractDataStorage>> storagesByRootId = dataStorageDao.loadDataStoragesByRootIds(
                     rootSearchResult.keySet()
-            ).stream().filter(
-                ds -> CollectionUtils.isEmpty(storageIdsToFilter) || storageIdsToFilter.contains(ds.getId())
-            ).collect(Collectors.groupingBy(AbstractDataStorage::getRootId));
+            ).stream()
+                    .filter(dataStorage -> permissionManager.storagePermission(dataStorage, READ_PERMISSION))
```

Comment:
I think we should avoid managing permissions on data storage manager layer if possible. It seems like something that DataStorageApiService should be doing f.e. by using `PostFilter`.

Reply:
I put it here because we need to filter available to read storage first, and that use that storage in search results. So If we move it to upper level (on api service) we may lose some of the results, because during the search will peek one that is not available.